src/clr-angular/data/datagrid/render/main-renderer.ts

```diff
@@ -118,6 +124,14 @@ export class DatagridMainRenderer<T = any> implements AfterContentInit, AfterVie
     this.rows.forEach(row => row.setColumnState());
   }
 
+  private findFirstVisible() {
+    this.headers.forEach(header => header.isFirstVisible());
```

Comment:
nit: for both function thing that `Array.map` is a better option.

Reply:
Why is map better? I am not trying to get a new array out of these headers. I just need to iterate through and *mutate* the items inside.

Reply:
ignore it .. sorry for that ..